osu.Game/Skinning/SkinnableTargetWrapper.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using Newtonsoft.Json;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// A container which is serialised and can encapsulate multiple skinnable elements into a single return type (for consumption via <see cref="ISkin.GetDrawableComponent"/>.
+    /// Will also optionally apply default cross-element layout dependencies when initialised from a non-deserialised source.
+    /// </summary>
+    [Serializable]
+    public class SkinnableTargetWrapper : Container, ISkinSerialisable
+    {
+        private readonly Action<Container> applyDefaults;
+
+        /// <summary>
+        /// Construct a wrapper with defaults that should be applied once.
+        /// </summary>
+        /// <param name="applyDefaults">A function with default to apply after the initial layout (ie. consuming autosize)</param>
```

Comment:
What does "consuming" autosize mean?

Reply:
For cases like score counters, they require a value to be filled via the bindable flow to get a size (via `SpriteText`'s `AutoSize`). Maybe badly worded here.

Reply:
Maybe just make it say "a function to apply the initial layout"? The line is pretty badly worded in general :P